packages/synapse-react-client/src/components/RowDataTable/RowDataTableWithQuery.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+  Query,
+  QueryBundleRequest,
+  Row,
+  SelectColumn,
+} from '@sage-bionetworks/synapse-types'
+import RowDataTable from './RowDataTable'
+import { LabelLinkConfig } from '../CardContainerLogic'
+import { parseEntityIdFromSqlStatement } from '../../utils/functions'
+import {
+  BUNDLE_MASK_QUERY_RESULTS,
+  BUNDLE_MASK_QUERY_SELECT_COLUMNS,
+} from '../../utils/SynapseConstants'
+import useGetQueryResultBundle from '../../synapse-queries/entity/useGetQueryResultBundle'
+import { SkeletonTable } from '../Skeleton'
+import { getFieldIndex } from '../../utils/functions/queryUtils'
+
+export type RowDataTableWithQueryProps = {
+  /** The query used to retrieve data */
+  query: Query
+  /** The specific row data to be rendered */
+  row: Row
+  /** The headers for the columns in the table */
+  headers: SelectColumn[]
```

Comment:
For this component, the query will retrieve the row and columns, so those should not be passed

```suggestion
```